firefox-ios/Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -294,8 +299,8 @@ class TabManagerImplementation: LegacyTabManager, Notifiable, WindowSimpleTabsPr
               let tabID = UUID(uuidString: tab.tabUUID)
         else { return }
 
-        Task {
-            await self.tabSessionStore.saveTabSession(tabID: tabID, sessionData: tabSession)
+        Task { [weak self] in
+            await self?.tabSessionStore.saveTabSession(tabID: tabID, sessionData: tabSession)
```

Comment:
Wasn't sure... does anyone foresee a case where we WOULD want to retain self here to ensure saving DOES complete? 🤔

Reply:
This should be fine. A save is triggered so often that even if it does deallocate before it finishes it's likely the previous most recent save is close enough the user won't notice.